ocr.py

```diff
@@ -44,3 +49,116 @@ def ocr(self, image_file, show_steps=False):
         features = self.feature_extractor.extract(image_file.image, segments)
         classes = self.classifier.classify(features)
         return classes, segments
+
+
+class SimpleOCR(object):
+    def __init__(self, segmentation="contour", extraction="simple", classification="knn",
+                 grounding="text"):
+        """
+        Makes the OCR process easier by providing all objects required for doing OCR
+        by itself, but allowing the user to choose other types of objects by specifying
+        them as strings in keyword arguments.
+
+        :param segmentation: contour, rawcontour or raw
+        :param extraction: simple (currently no other options are available)
+        :param classifier: knn (currently no other options are available)
+        :param grounding: text or user
+        """
+        if segmentation == "contour":
+            self.segmenter = segmenters.ContourSegmenter()
+        elif segmentation == "raw":
+            self.segmenter = segmenters.RawSegmenter()
+        elif segmentation == "rawcontour":
+            self.segmenter = segmenters.RawContourSegmenter()
+        else:
+            raise NotImplementedError
+        if extraction == "simple":
+            self.extractor = extractors.SimpleFeatureExtractor()
+        else:
+            raise NotImplementedError
+        if classification == "knn":
+            self.classifier = classifiers.KNNClassifier()
+        else:
+            raise NotImplementedError
+        if grounding == "text":
+            self.grounder = grounders.TextGrounder()
+        elif grounding == "user":
+            self.grounder = grounders.UserGrounder()
+        else:
+            raise NotImplementedError
+
+    def train(self, image_file, text=None):
+        """
+        feeds the training data to the OCR
+        :param image_file: The image file to be trained
+        :param text: Text to feed TextGrounder if this is the grounder type
+        :return:
+        """
+        if not image_file.is_grounded():
+            if not text and isinstance(self.grounder, grounders.TextGrounder):
+                raise ValueError("Ungrounded file with TextGrounder and text is None")
+            segments = self.segmenter.process(image_file.image)
+            if isinstance(self.grounder, grounders.TextGrounder):
+                self.grounder.ground(image_file.image, segments, text)
```

Comment:
If we have `ground()` decoupled from `set_text`, we can actually run it automatically in `OCR` as well - it's missing functionality at the moment. As long as we have clear exception messages, it should work well. I'll take a look into this, probably tomorrow